src/test/java/org/medicmobile/webapp/mobile/OpenSettingsDialogFragmentTest.java

```diff
@@ -59,6 +61,8 @@ public void setup() {
 		doNothing().when(openSettingsDialogFragment).startActivity(argsStartActivity.capture());
 
 		openSettingsDialogFragment.onCreate(null);
+		openSettingsDialogFragment.onActivityCreated(null);
+		shadowOf(Looper.getMainLooper()).idle();
```

Comment:
Is this test already covering the error when the fragment is unavailable, and do the retry?

Reply:
Not at the moment. I saw that the tests in this file were related to the behavior of the Fragment and decided not to add the test as it's an internal functionality. I could add it if it's needed.

Reply:
Is it possible to add a unit test to catch this bug in particular? It can be a good addition to prevent regressions and also as an example for similar future implementations

Reply:
The [setRetainInstance](https://developer.android.com/reference/androidx/fragment/app/Fragment.html#setRetainInstance(boolean)) is deprecated in androidx, this unit test can help the developer to consider that case when migrating to a more recent androidx lib.

Reply:
Sure, let me add a test for that.